app/scripts/models/auth_brokers/base.js

```diff
@@ -146,10 +149,38 @@ define([
      * for email verification. Useful for storing data that may be needed
      * by the verification tab.
      *
+     * @param {object} account
      * @return {promise}
      */
-    persist: function () {
-      return p();
+    persist: function (account) {
+      var self = this;
+
+      return p().then(function () {
+        // verification info is persisted to localStorage so that
+        // the same `context` is used if the user verifies in the same browser.
+        // If the user verifies in a different browser, the
+        // default (direct access) context will be used.
+        var verificationInfo =
+              createSameBrowserVerificationInfo(account, 'context');
+
+        verificationInfo.set({
+          context: self.relier.get('context')
+        });
+
+        return verificationInfo.persist();
+      });
+    },
+
+    /**
+     * Clear persisted verification data for the account
+     *
+     * @param {object} account
+     * @return {promise}
+     */
+    clearPersisted: function (account) {
```

Comment:
@vladikoff - I think `unpersistVerificationData` is definitely clearer than `clearPersisted`, perhaps we should update `persist` to be `persistVerificationData` while we are digging around. Thoughts?